### PR TITLE
add scnet

### PR DESCRIPTION
**Pull request by XinYangDong**
add scnet

### CODE REVIEW

File: scnet/train_oneflow.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+set -aux
+
+OFRECORD_PATH="./ofrecord/hustcolor"
```
Comment (Ldpe2G) [inline scnet/train_oneflow.sh:3]: 训练数据没提供下载链接

File: scnet/train_oneflow.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
+
+import oneflow.experimental as flow
```
Comment (Ldpe2G) [inline scnet/train_oneflow.py:2]: 最新的master分支已经没有 exprimental 了，可以更新到最新的代码试试

File: scnet/infer.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+set -aux
+
+PRETRAIN_MODEL_PATH="scnet_acc_0.947254"
+IMAGE_PATH="data/img_red.png"
```
Comment (Ldpe2G) [inline scnet/infer.sh:4]: image
这些图片都上传到阿里云，然后下载的方式，不要直接提交到github

File: scnet/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## Inference on Single Image
+### Download pretrained model
+pretrained [model](https://oneflow-public.oss-cn-beijing.aliyuncs.com/model_zoo/cv/classification/transfer_learning/scnet_acc_0.947254.zip) on hustcolor
+
+```bash
+bash infer.sh
+```
+ ![image](https://github.com/XinYangDong/models/blob/main/scnet/data/red_prediction.png)
+### Performer of model
```
Comment (Ldpe2G) [inline scnet/README.md:32]: ### Accracy

File: scnet/README.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# Scnet
+
+Scnet is a vehicle color classification network based on scloss, which can be easily combined with different backbone networks.
+
+
+
+## Train on [hustcolor](http://cloud.eic.hust.edu.cn:8071/~pchen/color.rar) Dataset
+
+### Prepare Traning Data
+If you use the vehicle color recognition dataset for testing your recognition algorithm you should try and make your results comparable to the results of others. We suggest to choose half of the images in each category to train a model. And use the other half images to test the recognition algorithm.
```
Comment (Ldpe2G) [inline scnet/README.md:10]: If you are using the vehicle color recognition dataset to test your recognition algorithm, you should try to make your results comparable to the others. We suggest using half of the images in each category to train a model and the other half for testing.

感觉表达可以在润色一下

File: scnet/README.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+
+### Run Oneflow Training script
+We have installed visdom to visualize the training model, and run the following program to enter http://localhost:8097/ get the training curve.
```
Comment (Ldpe2G) [inline scnet/README.md:14]: We have adpoted visdom to visualize the training process. Run the following command, then enter `http://localhost:8097/` to see the training curve.

表达润色一下

File: scnet/README.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+```
+python -m visdom.server
```
Comment (Ldpe2G) [inline scnet/README.md:17]: pip3 install -r requirements.txt --user
python -m visdom.server

File: scnet/utils/ofrecord_from_string.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import oneflow.core.record.record_pb2 as ofrecord
+import struct
+
```
Comment (Ldpe2G) [inline scnet/utils/ofrecord_from_string.py:3]: 这个脚本会用到吗

File: scnet/utils/plot.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
+import matplotlib.pyplot as plt
+
+
```
Comment (Ldpe2G) [inline scnet/utils/plot.py:3]: 这个脚本有用到吗

File: scnet/train_oneflow.py [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+
+    writer = open("of_losses.txt", "w")
+    for o in of_losses:
+        writer.write("%f\n" % o)
+    writer.close()
+
+if __name__ == "__main__":
+    args = _parse_args()
+    main(args)
+
```
Comment (Ldpe2G) [inline scnet/train_oneflow.py:139]: 下面有多余的空行

File: scnet/train_oneflow.py [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+        for b in range(len(train_data_loader)):
+            image, label = train_data_loader.get_batch()
+        
+            # oneflow train 
+            start_t = time.time()
+            image = image.to('cuda')
+            label = label.to('cuda')
+            logits,body = res50_module(image)
+            #loss = of_cross_entropy(logits, label)
```
Comment (Ldpe2G) [inline scnet/train_oneflow.py:95]: 删除无用的注释代码

File: scnet/models/resnet50.py [hunk 12]
```diff
@@ -0,0 +1,240 @@
+import oneflow.experimental as flow
```
Comment (Ldpe2G) [inline scnet/models/resnet50.py:1]: 不需要再 import oneflow.experimental as flow，改成 import oneflow as flow

File: scnet/README.md [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+bash train_oneflow.sh
+```
+## Inference on Single Image
+### Download pretrained model
+The pretrained [model](https://oneflow-public.oss-cn-beijing.aliyuncs.com/model_zoo/cv/classification/transfer_learning/scnet_acc_0.947254.zip) on hustcolor.
+
+```bash
+bash infer.sh
+```
+ ![image](https://github.com/XinYangDong/models/blob/main/scnet/data/red_prediction.png)
```
Comment (Ldpe2G) [inline scnet/README.md:27]: 这链接失效了，看不了

File: scnet/infer.sh [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+
+if [ ! -d "$IMAGE_PATH" ]; then
```
Comment (Ldpe2G) [inline scnet/infer.sh:6]: -d 是判断目录， -f 才是判断文件

File: scnet/train_oneflow.sh (hunk 1 above)
Comment (Ldpe2G) [inline scnet/train_oneflow.sh:3]: 把 https://oneflow-public.oss-cn-beijing.aliyuncs.com/datasets/models/scnet/ofrecord.zip 的下载也加到脚本里

**Review by Ldpe2G — APPROVED** [review]
